### ISSUE TITLE
Is it possible to support ESP32/8266 or RP2040?

### ISSUE THREAD

**Issue by megabitsenmzq** [swiftforarduino/community#12]
Both of these chips cost way less than Arduino. And many people are using them.  So it's good if you can support ESP32/8266 or RP2040.

**Comment by carlos4242** [comment]
Hi, thanks so much for the feedback! :) Arduino can be a bit expensive yes! What we recommend is directly buying the atmega328p microcontroller on its own, with a crystal and caps it's just a couple of $. Our product only supports AVR chips. We have found this approach has been easily cheap, powerful and flexible enough for hobbyists and professionals in our community using our IDE for their products. Plus much better power consumption and reduced footprint. No need for a $20 board! See our YouTube channel for some examples of how to do this.

**Comment by megabitsenmzq** [comment]
Got it! I'll try to build an Arduino from the chip! But still, it's not only about the price tag. ESP boards have onboard wifi and Bluetooth. Raspberry Pi Picos have a USB host feature. And they also have better performance. It's still a good thing to have. So if you can manage to support them in the future, it can be beneficial.

**Comment by carlos4242** [comment]
Yeah. Arm cortex support is definitely on the roadmap. Also riscV. Both amazing ISAs. The Tensilica ISA is a bit lower down the list tbh. Feels like Espressif's riscV chips are more the future! For usb host, look at atmega32u4. (For wifi/bt, we usually add modules to our projects where needed.) Drop us an email at swiftforarduino@gmail.com and I'll add you to our slack channels. We can start a lively discussion on all the above. We have some really fun (and helpful) chats!

**Comment by megabitsenmzq** [comment]
Wow, that sounds fun. I'll send a mail to you.
